akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala

```diff
@@ -232,6 +304,52 @@ final class EventEnvelope[Event](
    */
   def getTags(): JSet[String] = tags.asJava
 
+  def withPersistenceId(persistenceId: String): EventEnvelope[Event] =
```

Comment:
note to self: this should also take the `slice` as parameter since that must be updated together with the `persistenceId` and we can't calculate it here via the `Persistence(system)` extension.

The `entityType` can be extracted from the `persistenceId` and should also be updated.